cmd/ebrelayer/relayer/cosmos.go

```diff
@@ -30,17 +38,19 @@ type CosmosSub struct {
 	EthProvider             string
 	RegistryContractAddress common.Address
 	PrivateKey              *ecdsa.PrivateKey
+	CosmosContext           *types.CosmosContext
 	Logger                  tmLog.Logger
 }
 
 // NewCosmosSub initializes a new CosmosSub
 func NewCosmosSub(tmProvider, ethProvider string, registryContractAddress common.Address,
-	privateKey *ecdsa.PrivateKey, logger tmLog.Logger) CosmosSub {
+	privateKey *ecdsa.PrivateKey, cosmosContext *types.CosmosContext, logger tmLog.Logger) CosmosSub {
 	return CosmosSub{
 		TmProvider:              tmProvider,
 		EthProvider:             ethProvider,
 		RegistryContractAddress: registryContractAddress,
 		PrivateKey:              privateKey,
+		CosmosContext:           cosmosContext,
```

Comment:
Why do we need a cosmos context to initialize a new cosmos subscription?

Reply:
Cosmos context wraps up all information for send tx to cosmos, it will be used by both ethereum sub and cosmos sub.